mito-ai/src/Extensions/AppBuilder/auth-popup-deploy.tsx

```diff
@@ -0,0 +1,102 @@
+// auth-popup.tsx - Simple authentication popup
+import React from 'react';
+import { Authenticator } from '@aws-amplify/ui-react';
+import '@aws-amplify/ui-react/styles.css';
+import '../../../style/ConnectionForm.css';
+import '../../../style/button.css';
+import '../../../style/AuthPopup.css';
+import TextButton from '../../components/TextButton';
+
+interface AuthPopupProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSuccess: (user: any) => void;
+}
+
+// Using Mito's modal styling classes
+
+export const AuthPopup: React.FC<AuthPopupProps> = ({
+  isOpen,
+  onClose,
+  onSuccess
+}) => {
+  // Track if we've already called onSuccess to prevent infinite loops
+  const hasCalledOnSuccess = React.useRef(false);
+
+  if (!isOpen) return null;
+
+  return (
+    <div className="modal-overlay">
+      <div className="modal-content">
+        <div className="modal-header">
+          <h3>Sign In / Sign Up</h3>
+          <button
+            onClick={onClose}
+            className="modal-close-button"
+            title="Close"
+          >
+            ×
+          </button>
+        </div>
+
+        <Authenticator
```

Comment:
I'm noticing that after I sign in or Sign up, I'm always redirected to this screen. It doesn't make a ton of sense to be on this screen because I didn't just reset my password. Is that something we have control over? I don't see this in our code.. 

<img width="490" height="373" alt="Image" src="https://github.com/user-attachments/assets/bbb95a30-b928-4e38-998f-efef18a9bea6" />